docs: [Browser] Define the support browser runtimes (#5059)
open-telemetry/opentelemetry-js: README_NEXT.md

| Node.JS `v22`       | :heavy_check_mark:                            |
| Node.JS `v20`       | :heavy_check_mark:                            |
| Node.JS `v18`       | :heavy_check_mark:                            |
| Node.JS `v16`       | :heavy_check_mark:                            |
| Node.JS `v14`       | :heavy_check_mark:                            |
| Older Node Versions | See [Node Support](#node-support)             |
| Web Browsers        | See [Browser Support](#browser-support) below |

### Node Support

Only Node.js Active or Maintenance LTS versions are supported.
Previous versions of node *may* work, but they are not tested by OpenTelemetry and they are not guaranteed to work.
Note that versions of Node.JS v8 prior to `v8.12.0` will NOT work, because OpenTelemetry Node depends on the
`perf_hooks` module introduced in `v8.5.0` and `performance.timeOrigin` that is set correctly starting in `v8.12.0`.

### Browser Support

> [!IMPORTANT]
> Client instrumentation for the browser is **experimental** and mostly **unspecified**. If you are interested in
> helping out, get in touch with the [Client Instrumentation SIG][client-instrumentation-sig].

> Rather that define versions of specific browsers / runtimes, OpenTelemetry sets the minimum supported version based on the

Reviewer: ```suggestion
Rather than define versions of specific browsers / runtimes, OpenTelemetry sets the minimum supported version based on the
```

underlying Language features used.

The current minumum language feature support is set as [ECMAScript 2020](https://262.ecma-international.org/11.0/) that are available
in all modern browsers / runtimes.

This means that if you are targeting or your end-users are using a browser / runtime that does not support ES2020, you will need

> to transpile the code and provide any necessary polyfill's for the missing features to ensure compatibility with your target

Reviewer: ```suggestion
to transpile the code and provide any necessary polyfills for the missing features to ensure compatibility with your target
```

| [@opentelemetry/sdk-trace-node][otel-node]    | This module provides automatic tracing for Node.js applications. It is intended for use on the server only.                                                                                                                                           |
| [@opentelemetry/sdk-trace-web][otel-web]      | This module provides automated instrumentation and tracing for Web applications. It is intended for use in the browser only.                                                                                                                          |

### Compatible Exporters

OpenTelemetry is vendor-agnostic and can upload data to any backend with various exporter implementations. Even though, OpenTelemetry provides support for many backends, vendors/users can also implement their own exporters for proprietary and unofficially supported backends.

See the [OpenTelemetry registry](https://opentelemetry.io/registry/?language=js&component=exporter#) for a list of exporters available.

### Instrumentations

OpenTelemetry can collect tracing data automatically using instrumentations.

> To request automatic tracing support for a module not on this list, please [file an issue](https://github.com/open-telemetry/opentelemetry-js/issues). Alternatively, Vendor/Users can [write an instrumentation yourself](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/instrumentation-guide.md).

Reviewer: ```suggestion
To request automatic tracing support for a module not on this list, please [file an issue](https://github.com/open-telemetry/opentelemetry-js/issues). Alternatively, Vendors/Users can [write an instrumentation themselves](https://github.com/open-telemetry/opentelemetry-js/blob/main/doc/instrumentation-guide.md).
```
